File: server/lib/costCalculator.js
import MODELS from "./models.js";

const PLATFORM_FEE = 500;

function getModelCost(modelId, appType) {
  const list = MODELS[appType] || [];
  const model = list.find((m) => m.id === modelId);
  return model ? model.cost : 0;
}

function buildBudgetTiers(mockRow) {
  return {
    lean: {
      joules: mockRow.floor_joules + PLATFORM_FEE,
      label: "Lean MVP",
      desc: "Core features only",
      usd: ((mockRow.floor_joules + PLATFORM_FEE) * 0.0108).toFixed(2)
    },
    recommended: {
      joules: mockRow.market_joules + PLATFORM_FEE,
      label: "Recommended",
      desc: "Best balance of quality and speed",
      usd: ((mockRow.market_joules + PLATFORM_FEE) * 0.0108).toFixed(2)
    },
    full: {
      joules: Math.round(mockRow.market_joules * 1.15) + PLATFORM_FEE,
      label: "Full Scope",
      desc: "Maximum quality with buffer",
      usd: ((Math.round(mockRow.market_joules * 1.15) + PLATFORM_FEE) * 0.0108).toFixed(2)
    }
  };
}

export { getModelCost, buildBudgetTiers };

File: ai-service/services/cost_calculator.py
"""Cost calculator — RentPrompts costCalculator.js."""

from data.models import MODELS

PLATFORM_FEE = 500


def get_model_cost(model_id: str, app_type: str) -> float:
    models = MODELS.get(app_type, [])
    for model in models:
        if model.get("id") == model_id:
            return model.get("cost", 0)
    return 0


def build_budget_tiers(mock_row: dict) -> dict:
    floor_j = mock_row["floor_joules"]
    market_j = mock_row["market_joules"]
    return {
        "lean": {
            "joules": floor_j + PLATFORM_FEE,
            "label": "Lean MVP",
            "desc": "Core features only",
            "usd": f"{(floor_j + PLATFORM_FEE) * 0.0108:.2f}",
        },
        "recommended": {
            "joules": market_j + PLATFORM_FEE,
            "label": "Recommended",
            "desc": "Best balance of quality and speed",
            "usd": f"{(market_j + PLATFORM_FEE) * 0.0108:.2f}",
        },
        "full": {
            "joules": round(market_j * 1.15) + PLATFORM_FEE,
            "label": "Full Scope",
            "desc": "Maximum quality with buffer",
            "usd": f"{(round(market_j * 1.15) + PLATFORM_FEE) * 0.0108:.2f}",
        },
    }
